api/src/main/java/javax/enterprise/event/TransactionPhase.java

```diff
@@ -45,6 +49,15 @@
      * <p>
      * Identifies a before completion observer method, called during the before completion phase of the transaction.
      * </p>
+     * <p>
+     * It will also get invoked if
+     * <ul>
+     *     <li>there is no transaction in progress, or</li>
+     *     <li>the transaction already got rolled back, or</li>
```

Comment:
I would remove those these bullets with:

* a transaction is in progress but a JTA Synchronization callback cannot be registered (e.g. the transaction is marked for rollback)